ecgtools/parsers.py

```diff
@@ -92,3 +100,252 @@ def cmip6_default_parser(
+            )
+            return False
+        # verify that the first line is 'catalog:' and it only appears once in the yaml file
+        if len(self.input_yaml.keys()) != 1 or 'catalog' not in self.input_yaml.keys():
+            print(
+                "ERROR: The first line in the yaml file must be 'catalog:' and it should only appear once."
+            )
+            return False
+        if not isinstance(self.input_yaml['catalog'], list):
+            print(
+                'ERROR: The catalog entries are not in a list.  make sure you follow the corrrect format.'
+            )
+            return False
+        for dataset in self.input_yaml['catalog']:
+            # check to see if there is a data_sources key for each dataset
+            if 'data_sources' not in dataset.keys():
+                print("ERROR: Each experiment/dataset must have the key 'data_sources'.")
+                return False
+            # verify that we're working with a list at this level
+            if not isinstance(dataset['data_sources'], list):
+                print(
+                    'ERROR: The data_source entries are not in a list.  Make sure you follow the correct format.'
+                )
+                return False
+            for stream_info in dataset['data_sources']:
+                # check to make sure that there's a 'glob_string' key for each data_source
+                if 'glob_string' not in stream_info.keys():
+                    print("ERROR: Each data_source must contain a 'glob_string' key.")
+                    return False
+            # ensemble is an option, but we still need to verify that it meets the rules if it is added
+            if 'ensemble' in dataset.keys():
+                # verify that we're working with a list at this level
+                if not isinstance(dataset['ensemble'], list):
+                    print(
+                        'ERROR: The ensemble entries are not in a list.  Make sure you follow the correct format.'
+                    )
+                    return False
+                for stream_info in dataset['ensemble']:
+                    # check to make sure that there's a 'glob_string' key for each ensemble entry
+                    if 'glob_string' not in stream_info.keys():
+                        print("ERROR: Each ensemble must contain a 'glob_string' key.")
+                        return False
+        return True
+
+    def _validate_yaml(self):
+        """
+        Validates the generic yaml input against the schema.  It uses either yamale or the internal validater.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        boolean
+            True - passes the validation, False - fails the validation
+        """
+
+        # verify the format is correct
+        if self.validater == 'yamale':
+            try:
+                import yamale
```

Comment:
@sherimickelson, @andersy005:  Is there a reason why we don't just want to require `yamale` for YAML validation?  I don't think `yamale` is difficult to install, which means either (1) making `yamale` an installation dependency for `ecgtools` shouldn't be a problem or (2) making `yamale` an optional dependency, but required if you want to use the YAML parser.  

If you could require `yamale`, you could remove the internal validator all together.

Reply:
👍 for making `yamale` an installation dependency for `ecgtools`.

Reply:
@andersy005 and @kmpaul I can make yamale a dependency.